src/com/googlecode/jmxtrans/model/naming/StringUtils.java

```diff
@@ -40,4 +40,23 @@ public static String cleanupStr(String name, boolean allowDottedKeys) {
 		clean = SPACE_PAT.matcher(clean).replaceAll("");
 		return clean;
 	}
+
+	/**
+	 * Checks if a String is whitespace, empty (""), equals empty String ("\"\"") or null.
+	 *
+	 * <pre>
+	 * StringUtils.isBlank(null)      = true
+	 * StringUtils.isBlank("")        = true
+	 * StringUtils.isBlank(" ")       = true
+	 * StringUtils.isBlank("\"\"")    = true
+	 * StringUtils.isBlank("bob")     = false
+	 * StringUtils.isBlank("  bob  ") = false
+	 * </pre>
+	 *
+	 * @param name  the String to check, may be null
+	 * @return <code>true</code> if the String is null, empty, equals the empty string or whitespace
+	 */
+	public static boolean isBlankOrEmptyString(String name) {
+		return org.apache.commons.lang.StringUtils.isBlank(name) || "\"\"".equals(name);
```

Comment:
The underlying issue is actually that Yammer metrics itself is putting " characters into the Java ObjectName inside JMX. So I'm seeing things like: ObjectName="metric":type="",name="runtime"

You are right though that we might be able to clean it out earlier in the code.. I am not sure if this is a common issue with Yammer metrics (which appears to be pretty popular) or just the ones I'm looking at but in any case this is an edge case I'd like to fix without impacting anything else.

There is an existing StringUtils.cleanupStr method that runs pretty late in this whole process. Initially I thought the best thing would be for it to just strip trailing dots, but, if I do that it breaks this test in StringUtilsTests:

assertEquals("abcd.abcd.", StringUtils.cleanupStr("abcd.abcd.", true));

Just like KeyUtilsTests that @lookfirst commented on I'm not convinced that test case is valid, but, I don't really like breaking existing tests when I'm not sure if there was a good reason for them! :-)

I find this whole area inside KeyUtils to be fairly convoluted and probably needing a more major cleanup. As this is probably just an edge case though, I'm not sure if it is worth changing too much.